### PR TITLE
feat: Add v1 Qdrant/Splade_PP_en_v1

### PR DESCRIPTION
**Pull request by timonv**
Hey there, awesome work on this library. I'm enthousiastically using Fastembed.rs in Swiftide, but needed sparse vectors for hybrid search. This PR adds Splade to do just that.

### CODE REVIEW

**Comment by Anush008** [comment]
Hey.  How do you plan on using this? What would the implementation look like?

**Comment by Anush008** [comment]
We'd need a separate interface for sparse vectors.
Right now, only dense are supported.

**Comment by timonv** [comment]
Did I miss something? I thought this was all that was needed, the vectors looked alright locally. I want to generate sparse to do hybrid search with Splade, as I feel bm25 or cutoffs leave a lot on the table.

Looking over at the python implementation, is this the missing part? https://github.com/qdrant/fastembed/blob/9c72d2f59f91f87753da07c12a6f1082e233ecb3/fastembed/sparse/splade_pp.py#L39

That looks straightforward if ndarray supports the math.

**Comment by Anush008** [comment]
All the vectors are normalized.

**Comment by Anush008** [comment]
> > Looking over at the python implementation, is this the missing part? [qdrant/fastembed@9c72d2f/fastembed/sparse/splade_pp.py#L39](https://github.com/qdrant/fastembed/blob/9c72d2f59f91f87753da07c12a6f1082e233ecb3/fastembed/sparse/splade_pp.py#L39)

> That looks straightforward if ndarray supports the math.

Yes.

**Comment by timonv** [comment]
Cool, I'd be happy to pick it up

**Comment by Anush008** [comment]
For reference, https://github.com/Anush008/bm42-rs.

**Comment by timonv** [comment]
@Anush008 Ah cool, so if I understand correctly, the whole trick is to first normalize the weights, and then return all non zero values, and the indices for each weight into the token dictionary. And then I suppose it doesn't matter for a search engine what that dictionary is. That's pretty neat.

I've implemented it rather roughly by copying over most from text_embedding and tried to stick to the naming in the python version.

I *think* the math checks out. I haven't optimized or  otherwise cleaned anything up. Perhaps rayon would fit nicely into the post process step as well.

**Comment by Anush008** [comment]
Hey @timonv.
Awesome work and super quick turnaround. Thanks a lot.

Please feel free to add yourself to the authors list in the `Cargo.toml` file.

File: src/models/sparse.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SparseModel {
+    /// Splade PP Sparse vector model Qdrant/Splade_PP_en_v1
+    SPLADEPPV1,
+}
+
+pub fn models_list() -> Vec<ModelInfo<SparseModel>> {
+    vec![ModelInfo {
+        model: SparseModel::SPLADEPPV1,
+        dim: 30522,
```
Comment (Anush008) [inline src/models/sparse.rs:13]: Is this field needed?
How about a new info struct for sparse models without this field?

**Review by Anush008 — COMMENTED** [review]
Would you please add a reference in `README.md`?

File: src/models/sparse.rs (hunk 1 above)
Comment (timonv) [reply src/models/sparse.rs:13]: Ah no it's not, just felt it was cleaner to make the modelinfo generic. I could also make it 0 or wrap it in an option. What do you prefer?

Comment (Anush008) [reply src/models/sparse.rs:13]: We can't change the structure since it was public. Would be breaking for any consumer.

Comment (timonv) [reply src/models/sparse.rs]: Does the public api change when making it generic? I couldn't run semver-checks for some reason, haven't dived into it. I made it 0 for now, but happy to create a separate struct if needed.

Comment (Anush008) [reply src/models/sparse.rs]: I meant if we make it an `Option`.
Looks great now.

**Comment by timonv** [comment]
All right, looks like that's it from my side. Thanks for the hard work on this library!

**Comment by Anush008** [comment]
Thank you. I think this looks awesome enough to go out.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,10 @@ The default model is Flag Embedding, which is top of the [MTEB](https://huggingf
 - [**mixedbread-ai/mxbai-embed-large-v1**](https://huggingface.co/mixedbread-ai/mxbai-embed-large-v1)
 - [**Alibaba-NLP/gte-base-en-v1.5**](https://huggingface.co/Alibaba-NLP/gte-base-en-v1.5)
 
+## Sparse vectors
+
+- [**Qdrant/Splade_PP_en_v1**](https://huggingface.co/Qdrant/Splade_PP_en_v1) - Default
```
Comment (Anush008) [inline README.md:46]:
```suggestion
- [**prithivida/Splade_PP_en_v1**](https://huggingface.co/prithivida/Splade_PP_en_v1) - Default
```

Original author. Qdrant account contains the ONNX port.

**Pull request merged**